test/test_train_imagenet.py

```diff
@@ -180,12 +203,17 @@ def test_loop_fn(model, loader, device, context):
 
   accuracy = 0.0
   writer = SummaryWriter(log_dir=FLAGS.logdir) if FLAGS.logdir else None
+  num_distributed_machines = len(xm.xla_replication_devices(devices)) if len(devices) > 1 else 1
+  num_training_steps_per_epoch = len(train_dataset.imgs) // (
+      FLAGS.batch_size * (len(devices) or 1) * num_distributed_machines)
```

Comment:
That gives you devices, not machines.

Reply:
So `len(xm.xla_replication_devices(devices))` would give the total number of devices? Generally (8 * distributed world size)?

Reply:
This works as intended on non-distributed case using this code:
`
  num_devices = len(
      xm.xla_replication_devices(devices)) if len(devices) > 1 else 1
  num_training_steps_per_epoch = len(train_dataset.imgs) // (
      FLAGS.batch_size * num_devices)
`